lib/authz/project_purge/project_purge.go

```diff
@@ -0,0 +1,20 @@
+package project_purge
+
+import (
+	"context"
+	"math"
+	"time"
+)
+
+type PurgeClient interface {
+	PurgeProject(context.Context, string) error
+}
+
+func ExponentialNextCheck(numberOfFailures int) time.Time {
+	// use exponential backoff per retry of numberOfFailure^2 up until a max.
+	nextDuration := (time.Second * time.Duration(math.Pow(float64(numberOfFailures), 2.0)))
```

Comment:
Seems like the convention in the codebase is to prefer `1 << uint(tries)` instead of `math.Pow()`